### DISCUSSION TITLE
Echo (for 2.1.0)

### DISCUSSION THREAD

**Discussion by dc-redeclipse** [redeclipse/base#1214] [Maps and Content Showcase]
`Echo` is a re-light and ported version of the official team play map 'Echo' from Red Eclipse v1.6.
This map was original made by JoJo and W!CK3D and last remixed by Ulukai and was at last in Red Eclipse v1.6 official included in map rotation before it was excluded from the game since Red Eclipse v2.0.

**Echo of the Future** (New Version for Red Eclipse v2.1.0)
![echo](https://user-images.githubusercontent.com/60038236/229315989-663ab394-c15f-4435-9084-ca5a146821ca.png)

_New Version changes:_
- fixed blocks
- adjust and scale many block structures for a better gaming experience
- texture fixed and improved some places to look better and error correction
- reduce light elements and improved the old light settings
- I scale all the key elements up a bit to match the new game physics
- made new home bases because the old ones mismatch with the new re base model in bb in ctf
- add a descriptions and optimize the map size and terrain around
- hdrbright and sunlight improved
- optimized for dueling
- add rails and cameras
- enlarged the map with some more details around
- much higher clip
- eax sound
- cloud shadow
- and many other fixes of existing errors from the past versions

Big credits to @Jigoku @snipergoth @qreeves
and to all the other ones helping on that too.

_Ideal Gameplay:_
team & ffa deathmatch / capture the flag / bomberball / duel / onslaught / defend and  control / medieval
team & ffa race / gauntlet / lapped / onslaught / medieval
4-14 players or more

**Current Version**
**_RE 2.0.1_**
**current latest version = r1630**
[echo-r1630.zip](https://github.com/redeclipse/base/files/11130691/echo-r1630.zip)

**_RE 2.0.0_**
**old version = r1105**
[echo-r1105.zip](https://github.com/redeclipse/world/files/4239356/echo-r1105.zip)

_**History of Echo:**_
The original Echo map is based on the map 'Hollow' made by W!CK3D and known from Blood Frontier. First time the Echo map shows up in Red Eclipse v1.2 and was simply called 'Echo' and made by JoJo and Jonathan 'W!CK3D' Roels and stayed in game to Red Eclipse v1.3.
In Red Eclipse v1.4 the map was remixed by Ulukai and stayed in game to Red Eclipse v1.6 and had also the subtitle 'Echo of the Past'. So in conclusion this will be the only surviving map from Blood Frontier that still exist then in Red Eclipse in v2.0.1 and we hoping it will stay for the Future.

**Version History of old Echo maps:**
**_RE 1.6.0_**
**Echo of the Past (Ported for 2.0.0 and higher*)**
![echo-v1-6](https://user-images.githubusercontent.com/60038236/75280367-97092700-580d-11ea-8ef4-dcdb7e1f033d.png)
_Echo of the Past by JoJo and W!CK3D, remixed by Ulukai_
[echo-v1-6.zip](https://github.com/redeclipse/world/files/4250596/echo-v1-6.zip)

**_RE 1.2.0_**
**Echo (Ported for 2.0.0 and higher*)**
![echo-v1-2](https://user-images.githubusercontent.com/60038236/75280385-9e303500-580d-11ea-81b1-7a3724422099.png)
_Echo by JoJo and Jonathan 'W!CK3D' Roels_
[echo-v1-2.zip](https://github.com/redeclipse/world/files/4250598/echo-v1-2.zip)

**_Blood Frontier_**
![hollow](https://user-images.githubusercontent.com/60038236/75298153-05f87700-5832-11ea-8a5c-5c07ecc4f28a.png)
_Hollow by Jonathan 'W!CK3D' Roels_

_*Not guaranteed for the future but it currently works._

**Comment by dc-redeclipse** [comment]
**Impressions from `Echo of the Future`**
![20200222003912](https://user-images.githubusercontent.com/60038236/75080961-eb5e8f00-550d-11ea-8677-d7536213b747.png)
![20200222003835](https://user-images.githubusercontent.com/60038236/75080978-f7e2e780-550d-11ea-8856-3f69923d16b9.png)
![20200222003813](https://user-images.githubusercontent.com/60038236/75081005-04674000-550e-11ea-8796-bdac406b64f2.png)
![20200222003758](https://user-images.githubusercontent.com/60038236/75081024-10eb9880-550e-11ea-91f6-a4e88edd0a81.png)
![20200222003749](https://user-images.githubusercontent.com/60038236/75081032-1b0d9700-550e-11ea-87e7-6afea81ac4a5.png)
![20200222003736](https://user-images.githubusercontent.com/60038236/75081047-282a8600-550e-11ea-8f0e-c8c420e819c9.png)
![20200222003557](https://user-images.githubusercontent.com/60038236/75081057-34164800-550e-11ea-9aba-d089eed81ffb.png)
![20200222001319](https://user-images.githubusercontent.com/60038236/75081066-3c6e8300-550e-11ea-8fe0-b43fa5fedeef.png)
![20200222000246](https://user-images.githubusercontent.com/60038236/75081075-47c1ae80-550e-11ea-9880-992534185ec6.png)
![20200222000214](https://user-images.githubusercontent.com/60038236/75081092-50b28000-550e-11ea-9bbd-3ee7df8e044e.png)
![20200222000150](https://user-images.githubusercontent.com/60038236/75081107-5ad47e80-550e-11ea-8c38-e7f5701f8929.png)
![20200222000130](https://user-images.githubusercontent.com/60038236/75081122-67f16d80-550e-11ea-88c4-7145a5232b12.png)
![20200222000102](https://user-images.githubusercontent.com/60038236/75081132-750e5c80-550e-11ea-9455-310a741be368.png)
![20200221235946](https://user-images.githubusercontent.com/60038236/75081139-7f305b00-550e-11ea-9ccb-8fce9700b559.png)
![20200221235937](https://user-images.githubusercontent.com/60038236/75081151-89eaf000-550e-11ea-82ba-f5ea346778cf.png)
![20200221235929](https://user-images.githubusercontent.com/60038236/75081175-98d1a280-550e-11ea-812c-ad4f26b4d758.png)
![20200221235922](https://user-images.githubusercontent.com/60038236/75081193-a424ce00-550e-11ea-88c4-3b36cb6e629f.png)
![20200221235910](https://user-images.githubusercontent.com/60038236/75081214-b30b8080-550e-11ea-8f67-92a8698b232f.png)
![20200225125152](https://user-images.githubusercontent.com/60038236/75247888-3fe66080-57d3-11ea-971b-c22b2eeb4a55.png)
![20200225125127](https://user-images.githubusercontent.com/60038236/75247908-4e347c80-57d3-11ea-8dae-60120ea317dc.png)
![20200225125058](https://user-images.githubusercontent.com/60038236/75247917-54c2f400-57d3-11ea-85cf-4c5a6b6a084d.png)
![20200225124908](https://user-images.githubusercontent.com/60038236/75247935-60161f80-57d3-11ea-8701-8c26696ed7a9.png)

**Comment by qreeves** [comment]
I think you can do this: load up the broken version and at the highest grid size select the entire map, `entselect insel`, `entcopy`, load up the new map and at the highest grid size again select the entire map, `entselect insel`, `delent`, `entpaste`.

**Comment by cloudziq** [comment]
Was a bit more complicated, because on one version, he (dc) wanted to keep geometry and mapmodels, while lights + environment from another, so I just copied geometry + mapmodels from first to second, then replaced textures in cfg to match geometry I copied.  But no way to do this - since not all texture slots are in cfg (or it seems).  So I ended up with pasting everything else to map with nicer geometry, and set environmental settings more or less the same (less blueish fog, but this can be changed of course). 
Also, all lights need to relink now, I can help you dc with them - I saw that something is not right with links before anyway

before and after fix:
![20200210060146](https://user-images.githubusercontent.com/13541407/74122963-58e4f400-4bcd-11ea-8683-2e5c646054e1.jpg)
![20200210060115](https://user-images.githubusercontent.com/13541407/74122974-600c0200-4bcd-11ea-903a-6024ff1c4053.jpg)

fixed map: 
[echo_merged.zip](https://github.com/redeclipse/world/files/4179076/echo_merged.zip)

**Comment by dc-redeclipse** [comment]
**Updated to r742**
- big thanks to all who active pushing this in coop editing ahead!

**Comment by dc-redeclipse** [comment]
**Updated to r823**
- big thanks to @Dziq84 and Puzzle for there awesome help!

I find a wired looking in block shape in the tube @Jigoku made yesterday, but not sure if this is wanted, blo(a)ck magic, or actual a bug or error. You here what I mean: 
![20200212031131](https://user-images.githubusercontent.com/60038236/74300615-9c1ba000-4d50-11ea-9b41-33cc9d72bc67.png)
![20200212031149](https://user-images.githubusercontent.com/60038236/74300594-902fde00-4d50-11ea-878c-6f9b7933dbc4.png)
What is this?

**Comment by Jigoku** [comment]
Yeah i put those folded cubes there to get a smoother circle, because the cubes next to it are already on the smallest grid. If you just put a normal cube where they are, you'll see what the problem was :stuck_out_tongue: Can't get a smooth cylinder/circle shape otherwise. But they blend in with the texture/lighting so you can't see them on the outer edge.

**Comment by dc-redeclipse** [comment]
To do for next update:

- [x] make weapon pods for mines and grenades
- [x] put every playerstart, grunt and turret spawn 1/8 qube higher to avoid randomly falling down the map during spawn in game with resize and vampire both on
- [x] add more buildungs and fix the existing
- [x] improve the playerstart modes
- [x] add sand layer on some ground texture
- [x] fix brush and blendmap
- [x] fix textures inside the rooms, the down boarder offset
- [x] clips projectiles, ragdolls, cams, plasma and fire passing glass

**Comment by dc-redeclipse** [comment]
**Updated to r823**
- big thanks to @Dziq84 for the awesome help again!

**Comment by cloudziq** [comment]
not sure if you like the result since grass can't be nicely blended currently, also bush4 which you want to keep, can't have altered color (which I going to fix now)

updated:  [echo.zip](https://github.com/redeclipse/world/files/4196912/echo.zip)

**Comment by Jigoku** [comment]
> grass can't be nicely blended currently

looks nicely blended to me :thumbsup:

**Comment by dc-redeclipse** [comment]
**Updated to r924**
- add grass > big thanks to @Dziq84 for the nice blendmap grass!
- add a little Easter egg with a bit cube scripting

**Comment by dc-redeclipse** [comment]
**Updated to r985**
- finish the buildings around
- many little things fixed
! texturecull must be done at once I know but for debugging all texture are on there for that!

**Comment by dc-redeclipse** [comment]
**Updated to r1075**
- improved all playerstarts again
- many other little things fixed
- make all waypoints new
- little terrain improvement in center area
- put decals in center area
- texturecull

**Comment by dc-redeclipse** [comment]
**Updated to r1098**
- every hallway is now enlarged
- clip celling is now much higher and bigger
- no-clip on tube is now bigger
- all lights on weapon pods fixed
- removed wired alpha in the sky
- many other little things fixed again
- switch rocket in the air above only in bomber and the drain rocket for all other game variants
- texturecull

**Comment by dc-redeclipse** [comment]
**Updated to r1100**
- anti glitching fix
- bigger clip on weak points

**Comment by dc-redeclipse** [comment]
**Updated to r1105**
- more anti glitching fix

**Comment by dc-redeclipse** [comment]
**Updated to r1182**
- add rails for mapmodels and  cameras
- bug fixes lights and texture
- many very little things improved

**Comment by dc-redeclipse** [comment]
**Updated to r1184**
- add extra roller and drone spawn

**Comment by dc-redeclipse** [comment]
_**History of Echo:**_
The original Echo map is based on the map 'Hollow' made by W!CK3D and known from Blood Frontier. First time the Echo map shows up in Red Eclipse v1.2 and was simply called 'Echo' and made by JoJo and Jonathan 'W!CK3D' Roels and stayed in game to Red Eclipse v1.3.
In Red Eclipse v1.4 the map was remixed by Ulukai and stayed in game to Red Eclipse v1.6 and had also the subtitle 'Echo of the Past'. So in conclusion this will be the only surviving map from Blood Frontier that still exist then in Red Eclipse in v2.0.1 and we hoping it will stay for the Future.

**Version History of old Echo maps:**
**_RE 1.6.0_**
**Echo of the Past (Ported for 2.0.0 and higher*)**
![echo-v1-6](https://user-images.githubusercontent.com/60038236/75280367-97092700-580d-11ea-8ef4-dcdb7e1f033d.png)
_Echo of the Past by JoJo and W!CK3D, remixed by Ulukai_
[echo-v1-6.zip](https://github.com/redeclipse/world/files/4250596/echo-v1-6.zip)

**_RE 1.2.0_**
**Echo (Ported for 2.0.0 and higher*)**
![echo-v1-2](https://user-images.githubusercontent.com/60038236/75280385-9e303500-580d-11ea-81b1-7a3724422099.png)
_Echo by JoJo and Jonathan 'W!CK3D' Roels_
[echo-v1-2.zip](https://github.com/redeclipse/world/files/4250598/echo-v1-2.zip)

**_Blood Frontier_**
![hollow](https://user-images.githubusercontent.com/60038236/75298153-05f87700-5832-11ea-8a5c-5c07ecc4f28a.png)
_Hollow by Jonathan 'W!CK3D' Roels_

_*Not guaranteed for the future but it currently works._

**Comment by dc-redeclipse** [comment]
**Updated to r1226**
- fix all camera flags
- improve mapmodel movement
- place mapshot camera
- little texture fix on one building

**Comment by dc-redeclipse** [comment]
**Updated to r1254**
- remove unneeded blendmap (big thanks to @Jigoku for noticing that)
- improve grunt spawns
- place mapshot camera a bit higher
- little texture fix

**Comment by dc-redeclipse** [comment]
**Updated to r1267**
- fix ship rotation to fit the new game defaults
- camera improvements
- little texture fix

**Comment by dc-redeclipse** [comment]
**Updated to r1306**
- camera improvements
- little texture fixes again
- improve mapshot cam place

**Comment by dc-redeclipse** [comment]
**Updated to r1350**
- set atmodensity
- little texture fixes again
- improve mapshot cam place

**Comment by dc-redeclipse** [comment]
**Updated to r1358**
- set atmodensity back to and fixatmo
- little texture fixes again
- reduce cams and rails to improve spec cam selection

**Comment by dc-redeclipse** [comment]
**Updated to r1360**
- little texture fixes
- fix mapmodel path in cfg for snipergoth/cable/*

**Comment by dc-redeclipse** [comment]
**Updated to r1364**
- little texture and geo fixes on the tube
- improved clip material on the walls

**Comment by dc-redeclipse** [comment]
**Updated echo to r1371**
- some texture and geo fixes
- add some decals
- improved clip material on some places

**Comment by dc-redeclipse** [comment]
**Updated echo to r1477**
- little texture fixes
- cleanup
- new positioning of the mapshot camera

**Comment by dc-redeclipse** [comment]
**Updated echo to r1478**
- little texture fixes
- cleanup

**Comment by dc-redeclipse** [comment]
**Updated echo to r1610**
- many texture alignment fixes
- eax sound implementation
- fix spawnpoints
- improved outer areas
- improved light colors
- fixed cam rails
- cleanup

**Comment by dc-redeclipse** [comment]
**Updated echo to r1611**
- small texture fixes
- eax sound fixes

**Comment by dc-redeclipse** [comment]
**Updated echo to r1613**
- small light fix

**Comment by dc-redeclipse** [comment]
**Updated echo to r1625**
- some more texture fixes and improved alignments

**Comment by dc-redeclipse** [comment]
**Updated echo to r1630**
- fix player and actor spawn position height
